File: 알고리즘/CodingTest/src/Programmers/level1/Sqrt.java
package Programmers.level1;
import java.lang.Math;

// https://programmers.co.kr/learn/courses/30/lessons/12934

public class Sqrt {
    public long solution(long n) {
        long res = (long)(Math.sqrt(n));

        if(n == Math.pow(res, 2)) { // key: 제곱근과 제곱된 값을 비교
            return (long)Math.pow(res + 1 , 2);
        } else {
            return -1;
        }
    }
}

File: Algorithm/Algorithm_py/programmers1/Sqrt.py
# https://programmers.co.kr/learn/courses/30/lessons/12934?language=python3

def solution(n):
  answer = -1
  x = n**(1/2)
  
  if x == int(x):
    # 제곱근의 정수 부분만을 비교(음수나 3,5 등의 경우는 소수점으로 나오기 때문)
    answer = (x+1)**2
  return answer

print(solution(121))